File: modules/libsignal/android/src/main/java/expo/modules/libsignal/LibSignalModule.kt
package expo.modules.libsignal

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/** Native shell; cryptographic operations use the TypeScript Signal engine. */
class LibSignalModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("LibSignalModule")
    Function("isNativeShell") {
      true
    }
  }
}

File: modules/libsignal/ios/LibSignalModule.swift
import ExpoModulesCore

/// Native shell for the libsignal Expo module. Crypto runs in the TS layer (Hermes).
public class LibSignalModule: Module {
  public func definition() -> ModuleDefinition {
    Name("LibSignalModule")
    Function("isNativeShell") { () -> Bool in
      true
    }
  }
}
